### PR TITLE
Tesla Update

### PR DESCRIPTION
**Pull request by silicons**
:cl:
experimental: Tesla damage has been reworked to structures and objects have been worked with a linear scaling.
experimental: All tesla weapons now no longer need to hit a mob to arc lightning; however, they will no longer stun, and only hit each target once.
/:cl:

tesla_zap vars are now a single bitfield.

### CODE REVIEW

**Comment by Vidiveni** [comment]
Will the tesla shock from teslium/water grenades still stun?

**Comment by kevinz000** [comment]
yes

File: code/game/machinery/_machinery.dm [hunk 1]
```diff
@@ -463,15 +463,14 @@ Class Procs:
...
+/obj/machinery/tesla_act(power, tesla_flags, shocked_objects)
 	..()
-	if(prob(85) && explosive)
-		explosion(src.loc, 1, 2, 4, flame_range = 2, adminlog = FALSE, smoke = FALSE)
-	else if(prob(50))
-		emp_act(EMP_LIGHT)
-	else
-		ex_act(EXPLODE_HEAVY)
+	if(prob(85) && (tesla_flags & TESLA_MACHINE_EXPLOSIVE))
+		explosion(loc, 1, 2, 4, flame_range = 2, adminlog = FALSE, smoke = FALSE)
```
Comment (Cyberboss) [inline code/game/machinery/_machinery.dm]: loc => src

**Review by Cyberboss — CHANGES_REQUESTED** [review]

**Comment by optimumtact** [comment]
stale kevin pr spam

File: code/modules/clothing/suits/reactive_armour.dm [hunk 2]
```diff
@@ -150,12 +150,21 @@
...
-	var/tesla_boom = FALSE
-	var/tesla_stun = FALSE
+	var/tesla_flags = TESLA_MOB_DAMAGE | TESLA_OBJ_DAMAGE
+
+/obj/item/clothing/suit/armor/reactive/tesla/dropped(mob/user)
+	..()
+	if(istype(user))
+		user.flags_1 &= ~TESLA_IGNORE_1
+
+/obj/item/clothing/suit/armor/reactive/tesla/equipped(mob/user, slot)
```
Comment (kevinz000) [inline code/modules/clothing/suits/reactive_armour.dm]: going to point this out so no one screeches. it prevents tesla armor nerds from getting targeted by tesla bolts. was in the original pr but people might have different opinions now that tesla armor is a thing.

**Pull request merged**